internal/api/promote_downstream_v1alpha1.go

```diff
@@ -92,22 +92,26 @@ func (s *server) PromoteDownstream(
 		}
 		return nil, connect.NewError(connect.CodeNotFound, err)
 	}
-	if !s.isFreightAvailableFn(
-		freight,
-		"",                  // approved for not considered
-		[]string{stageName}, // verified in
-	) {
```

Comment:
Awkward usage of `IsFreightAvailable()`, which was written awkwardly to accommodate this.